packages/noya-state/src/selectors/layerSelectors.ts

```diff
@@ -95,3 +96,21 @@ export const addSiblingLayer = <
   const l = layer instanceof Array ? layer : [layer];
   parent.layers.splice(indexPath[indexPath.length - 1], 0, ...l);
 };
+
+export const getLeftMostLayerPoint = (page: Sketch.Page) => {
```

Comment:
For this function, I'd recommend sorting by `layer.frame.x` and choosing the first layer in the sorted array. I think the logic will be simpler